src/highdicom/legacy/sop.py

```diff
@@ -25,384 +31,506 @@
-        tag_for_keyword('SourceImageSequence'),
-        tag_for_keyword('BodyPartExamined'),
-        tag_for_keyword('IrradiationEventUID'),
-        tag_for_keyword('RescaleIntercept'),
-        tag_for_keyword('RescaleSlope'),
-        tag_for_keyword('RescaleType'),
-    }
-
-    if ref_ds.ImageType[0] == 'ORIGINAL':
-        mf_dataset.VolumeBasedCalculationTechnique = 'NONE'
-    else:
-        mf_dataset.VolumeBasedCalculationTechnique = 'MIXED'
-
-    pixel_representation = sf_datasets[0].PixelRepresentation
-    volumetric_properties = 'VOLUME'
-    unique_image_types = set()
-    unassigned_dataelements: Dict[str, List[Dataset]] = defaultdict(list)
-
-    # Per-Frame Functional Groups
-    perframe_items = []
-    for i, ds in enumerate(sf_datasets):
-        perframe_item = Dataset()
-
-        # Frame Content (M)
-        frame_content_item = Dataset()
-        if 'AcquisitionDate' in ds and 'AcquisitionTime' in ds:
-            frame_content_item.FrameAcquisitionDateTime = '{}{}'.format(
-                ds.AcquisitionDate,
-                ds.AcquisitionTime
-            )
-        frame_content_item.FrameAcquisitionNumber = ds.InstanceNumber
-        perframe_item.FrameContentSequence = [
-            frame_content_item,
-        ]
+    @classmethod
+    def istag_repeating_group(cls, t: BaseTag) -> bool:
+        g = t.group
+        return (g >= 0x5000 and g <= 0x501e) or\
+            (g >= 0x6000 and g <= 0x601e)
 
-        # Plane Position (Patient) (M)
-        plane_position_item = Dataset()
-        plane_position_item.ImagePositionPatient = ds.ImagePositionPatient
-        perframe_item.PlanePositionSequence = [
-            plane_position_item,
-        ]
+    @classmethod
+    def istag_group_length(cls, t: BaseTag) -> bool:
+        return t.element == 0
 
-        frame_type = list(ds.ImageType)
-        if len(frame_type) < 4:
-            if frame_type[0] == 'ORIGINAL':
-                frame_type.append('NONE')
-            else:
-                logger.warn('unknown derived pixel contrast')
-                frame_type.append('OTHER')
-        unique_image_types.add(tuple(frame_type))
-        frame_type_item = Dataset()
-        frame_type_item.FrameType = frame_type
-        frame_type_item.PixelRepresentation = pixel_representation
-        frame_type_item.VolumetricProperties = volumetric_properties
-        if frame_type[0] == 'ORIGINAL':
-            frame_type_item.FrameVolumeBasedCalculationTechnique = 'NONE'
+    @classmethod
+    def isequal(cls, v1: Any, v2: Any) -> bool:
+        from pydicom.valuerep import DSfloat
+        float_tolerance = 1.0e-5
+
+        def is_equal_float(x1: float, x2: float) -> bool:
+            return abs(x1 - x2) < float_tolerance
+        if type(v1) != type(v2):
+            return False
+        if isinstance(v1, DataElementSequence):
+            for item1, item2 in zip(v1, v2):
+                cls.isequal_dicom_dataset(item1, item2)
```

Comment:
Is this correct - shouldn't the return value for this function be checked?

Reply:
You are right, corrected.